gateway/src/resty/mime.lua

```diff
@@ -0,0 +1,39 @@
+local format = string.format
+local lower = string.lower
+local setmetatable = setmetatable
+
+local re = require('ngx.re')
+local re_match = ngx.re.match
+local re_split = re.split
+
+local _M = {}
+local MimeType_mt = { __index = _M }
+
+
+function _M.new(media_type)
+  -- The [RFC[(https://tools.ietf.org/html/rfc7231#section-3.1.1.1) defines
+
+  -- that space around `;` is irrelevant, so remove it when splitting
+  local match = re_split(media_type, [[\s*;\s*]], 'oj', nil, 2)
+
+  local self = {}
+
+  -- that the type can include upper and lower-case chars.
+  -- Let's convert it to lower-case for easier comparisons.
+  self.media_type = lower(match[1])
+  self.parameters = match[2]
+
+  return setmetatable(self, MimeType_mt)
+end
+
+function _M:parameter(name)
+  local parameters = self.parameters
+
+  local matches = re_match(parameters, format([[%s=(?:"(.+?)"|([^\s;"]+))\s*(?:;|$)]], name), 'oji')
```

Comment:
How about extracting that regex into its own var?
Maybe even explain in a line or two its purpose.

Reply:
Done.